src/fragments/lib/push-notifications/react-native/migrate_from_previous_version/migrate-from-previous-version.mdx

```diff
@@ -0,0 +1,142 @@
+
+```groovy
+buildscript {
+  ext {
+    ...
+    minSdkVersion = 24
+    ...
+  }
+}
+```
+
+3. Open the application `build.gradle` located inside the `/android/app` folder of your React Native project with a text editor.
+
+4. Remove firebase dependencies from the `dependencies` block.
+
+```diff
+dependencies {
+  ...
+-   implementation "com.google.firebase:firebase-core:15.0.2"
+-   implementation "com.google.firebase:firebase-messaging:15.0.2"
+  ...
+}
+```
+
+### Update the manifest
+
+5. Open `AndroidManifest.xml` located inside the `/android/app/src/main` folder.
+
+6. Remove the old service block from the `application`.
+
+```diff
+<application ... >
+  ...
+- <!--[START Push notification config -->
+- <service
+-   android:name="com.amazonaws.amplify.pushnotification.RNPushNotificationMessagingService"
+-   android:exported="false">
+-   <intent-filter>
+-     <action android:name="com.google.firebase.MESSAGING_EVENT"/>
+-   </intent-filter>
+- </service>
+- <!-- [END Push notification config -->
+  ...
+</application>
+```
+
+</Block>
+</BlockSwitcher>
+
+## Update your application code
+
+### Add required polyfills
+
+You need to add the crypto.getRandomValues and URL polyfill to your application's entry point file (in most React Native apps this will be the top level `index.js`).
+
+```js
+// Example index.js
+import 'react-native-get-random-values';
+import 'react-native-url-polyfill/auto';
+...
+```
+
+### Remove Analytics (Optional)
+
+**If you do not** use Amplify Analytics in your application but have previously imported it in your code just to implement Amplify Push Notification, you should be able to remove it now. Analytics is no longer a required to be configured in your frontend application to use Push Notification.
+
+### Replace APIs
+
+Please refer to the [Getting started](/lib/push-notifications/getting-started) portion of this guide to start learning more about the new Amplify Push Notifications.
+
+A brief, but imperfect, mapping of your APIs to their replacements is as follows:
+
+|      Previous API       |                                  Replaced by                                   |
+| :---------------------: | :----------------------------------------------------------------------------: |
+|      `onRegister`       |                               `onTokenReceived`                                |
+| `onNotificationOpened`  |                             `onNotificationOpened`                             |
+|    `onNotification`     | `onNotificationReceivedInForeground`<br />`onNotificationReceivedInBackground` |
+| `requestIOSPermissions` |                              `requestPermissions`                              |
+
+We _strongly_ recommend you take the time to explore all the ways the new version of Amplify Push Notification are different and all the new features that have been added. This will help you to make the best decision about how best to adopt these changes for your unique use cases.
```

Comment:
```suggestion
We _strongly_ recommend you take the time to explore all the ways the new version of Amplify Push Notification are different and all the new features that have been added. This will help you to make the best decision about how to adopt these changes for your unique use cases.
```

"best" twice in one sentence maybe it should be `This will help you to make the decision about how best to adopt these changes for your unique use cases` 
or alternatively `This will help you to make the best decision about how to adopt these changes for your unique use cases.` 
I think I like the 2nd one better but feel free to argue this one